runtime/doc/lsp.txt

```diff
@@ -1469,6 +1462,63 @@ save({lenses}, {bufnr}, {client_id})                 *vim.lsp.codelens.save()*
       • {client_id}  (integer)
 
 
+==============================================================================
+Lua module: vim.lsp.inlay_hint                                *lsp-inlay_hint*
+
+get({opts})                                         *vim.lsp.inlay_hint.get()*
+    Get the list of inlay hints, (optionally) restricted by buffer, client, or
+    range.
+
+    Example usage: >lua
+        local timeout, bufnr = 100, vim.api.nvim_get_current_buf()
+        local hint = vim.lsp.inlay_hint.get({ bufnr = bufnr })[1]
+
+        local client = vim.lsp.get_client_by_id(hint.client_id)
+        resolved_hint = client.request_sync(
+          'inlayHint/resolve', hint.inlay_hint, timeout, bufnr).result
+        vim.lsp.util.apply_text_edits(resolved_hint.textEdits, bufnr, client.encoding)
+
+        location = resolved_hint.label[1].location
+        client.request("textDocument/hover", {
+          textDocument = { uri = location.uri },
+          position = location.range.start,
+        })
+<
+
+    Parameters: ~
+      • {opts}  vim.lsp.inlay_hint.get.opts ? Optional table which holds the following optional fields:
+                • bufnr (integer?): Only return inlay hints from the clients
+                  attached to the given buffer.
+                • filter (function?): Predicate used to filter clients.
+                  Receives a client as argument and must return a boolean.
+                  Clients matching the predicate are included. Example: >lua
+
+                     -- Don't request inlay hints from typescript-language-server
+                     vim.lsp.inlay_hint.get {
+                       filter = function(client) return client.name ~= "tsserver" end
```

Comment:
At this point we don't have hint objects yet, we have to loop through the each client, then through each line for that client, then for each hint of that line. This is for cutting off the first loop

Reply:
It's definitely a weird name though as it's something that seems to imply filtering in post

(and filtering in post wouldn't be a necessary feature, as users can just `vim.tbl_filter`)

Reply:
Actually we should just remove this field entirely I think lol. I pasta'd it from `vim.lsp.buf.format` but I'm actually not sure what value it provides.

Reply:
Is this field actually needed? What other Nvim lsp features support client filtering?

Reply:
Request-specific methods like `vim.lsp.buf.format` and `vim.lsp.buf.rename` have a similar `filter` field.

Reply:
> Request-specific methods like vim.lsp.buf.format and vim.lsp.buf.rename have a similar filter field.

They have a filter because they have a side effect that depends on the clients, not a result that one could post-process.

If there's no big performance advantage from moving the filters into the method I'd also tend to remove or trim them

Reply:
The range is the important filter due to the how the inlay hint data structure is (array per line - which we could revisit if it became an issue)

Bufnr also seems important if someome has a lot of buffers

But client, yeah i think we can remove